sklearnex/tests/test_common.py

```diff
@@ -52,3 +57,38 @@ def test_target_offload_ban():
 
     output = "\n".join(output)
     assert output == "", f"sklearn versioning is occuring in: \n{output}"
+
+
+def _sklearnex_walk(func):
+    """this replaces checks on pkgutils to look through sklearnex
+    folders specifically"""
+
+    def wrap(*args, **kwargs):
+        if "prefix" in kwargs and kwargs["prefix"] == "sklearn.":
+            kwargs["prefix"] = "sklearnex."
+        if "path" in kwargs:
+            # force root to sklearnex
+            kwargs["path"] = [str(pathlib.Path(__file__).parent.parent)]
+        for pkginfo in func(*args, **kwargs):
+            # Do not allow spmd to be yielded
+            if "spmd" not in pkginfo.name.split("."):
+                yield pkginfo
+
+    return wrap
+
+
+def test_all_estimators_covered(monkeypatch):
+    """check that all estimators defined in sklearnex are available in either the
+    patch map or covered in special testing via SPECIAL_INSTANCES. The estimator
+    must inherit sklearn's BaseEstimator and must not have a leading underscore.
+    The sklearnex.spmd and sklearnex.preview packages are not tested.
+    """
+    monkeypatch.setattr(pkgutil, "walk_packages", _sklearnex_walk(pkgutil.walk_packages))
+    estimators = all_estimators()
+    print(estimators)
+    for name, obj in estimators:
+        # do nothing if defined in preview
+        if "preview" not in obj.__module__:
```

Comment:
Preview has not been centrally tested up to this point. It would also conflict with the PATCHED_MODELS, as we would need to then bookkeep for two versions of the same estimator throughout testing.  Individual preview tests are discovered by pytest, but not discovered in a meaningful way by sklearnex.

Reply:
We can enable preview via env variable in this case they should be centrally tested for patched models don't they?

Reply:
True, luckily so far we don't do that in any of the CI systems.  I guess it raises questions about what defines a preview estimator, I assumed it was because of code-quality and/or performance.